### PR TITLE
Change widget layout to place in last column

### PR DESCRIPTION
**Pull request by AlexanderWells-diamond**
Now new group and widget placement behaves the same: They'll be placed in the rightmost column if they fit, or into a new column. This means the order of rendered widgets will exactly match the input order - we no longer reshuffle in an attempt to preserve screen space.

Previously new groups would be placed into the leftmost column where it would fit, or placed into the last column.
Standalone widgets would be placed in either of the last two columns, if it would fit.

### CODE REVIEW

File: src/pvi/_format/utils.py [hunk 1]
```diff
@@ -317,45 +317,35 @@ def screen(self, components: Tree[Component], title: str) -> WidgetFactory[T]:
...
+                    column_bounds=last_column_bounds,
                     parent_bounds=screen_bounds,
-                    next_column_bounds=columns[-1],  # Last column
+                    next_column_bounds=next_column_bounds,
                     group_widget_indent=self.layout.group_widget_indent,
                 )
 
-            if columns[-1].y != 0:
+            if next_column_bounds.y != 0:
                 # We added to the last column - make a new empty column
```
Comment (GDYendell) [inline src/pvi/_format/utils.py:347]: I think this comment is unnecessary now

File: src/pvi/_format/utils.py [hunk 2]
```diff
@@ -317,45 +317,35 @@ def screen(self, components: Tree[Component], title: str) -> WidgetFactory[T]:
...
-                    )
-
-                    if group.bounds.h + group.bounds.y <= screen_bounds.h:
-                        # Group fits in this column
-                        break
-
-                group.bounds.w += self.layout.group_width_offset
-                screen_widgets.append(group)
+                # Note: Group adjusts bounds to fit the components
+                screen_widgets = screen_widgets + self.make_group_widget(
```
Comment (GDYendell) [inline src/pvi/_format/utils.py:329]: Could `make_group_widget` not return a list so it can just do an append here?

Comment (AlexanderWells-diamond) [reply src/pvi/_format/utils.py:329]: `make_group_widget` does return a list - I made it look very similar to `make_component_widgets`, which uses this syntax a few lines below. Would you prefer I update both to use `.append()`?

File: src/pvi/_format/utils.py (hunk 1 above)
Comment (AlexanderWells-diamond) [reply src/pvi/_format/utils.py:347]: Agreed, I'll remove it.

File: src/pvi/_format/utils.py (hunk 2 above)
Comment (GDYendell) [reply src/pvi/_format/utils.py:329]: Ah OK. That is reasonable. In that case can you make both usages call `extend`?

File: src/pvi/_format/utils.py (hunk 1 above)
Comment (AlexanderWells-diamond) [reply src/pvi/_format/utils.py]: Done in https://github.com/epics-containers/pvi/pull/19/commits/74d8d6838be3d1256454efe0db773af4259db620

File: src/pvi/_format/utils.py (hunk 2 above)
Comment (AlexanderWells-diamond) [reply src/pvi/_format/utils.py]: Done in https://github.com/epics-containers/pvi/pull/19/commits/74d8d6838be3d1256454efe0db773af4259db620

**Review by GDYendell — APPROVED** [review]

**Pull request merged**